modular_citadel/code/modules/reagents/reagent container/hypospraymkii.dm

```diff
@@ -1,6 +1,21 @@
 #define HYPO_SPRAY 0
 #define HYPO_INJECT 1
 
+#define WAIT_SPRAY 50
+#define WAIT_INJECT 40
+#define SELF_SPRAY 20
+#define SELF_INJECT 10
+
+#define DELUXE_WAIT_SPRAY 25
+#define DELUXE_WAIT_INJECT 20
+#define DELUXE_SELF_SPRAY 10
+#define DELUXE_SELF_INJECT 0
+
+#define COMBAT_WAIT_SPRAY 0
```

Comment:
0 durations still fuck with do_after.  and I'm still not convinced self application needs to be this much faster for the others.

Reply:
You're free to change the numbers around once merged. My tests show no problems with 0-duration do_mob(). However, I hold that self-application should be faster than application to someone else, that injections should be faster than sprays, and that the CMO should have instant self injections. (Obviously, combat has instant everything).

Reply:
Also, keep in mind that your original default number for everything was 3 seconds, since the exact delay wasn't specified.

Reply:
https://github.com/Citadel-Station-13/Citadel-Station-13/blob/ce0b0127a99dd12c3c56878fc36ba0e1fbefbcc1/code/__HELPERS/mobs.dm#L339

do_mob shouldn't break with zero because of this check.

Reply:
shouldn't doesn't mean wouldn't. :P

Reply:
The game is single-threaded, so in theory, it literally can't.